lib/src/source_visitor.dart

```diff
@@ -1102,7 +1103,183 @@ class SourceVisitor extends ThrowingAstVisitor {
+        ..previous = operator.previous
+        ..next = operator.next;
+
+  static Expression _realTargetOf(Expression expression) {
+    if (expression is PropertyAccess) {
+      return expression.realTarget;
+    } else if (expression is MethodInvocation) {
+      return expression.realTarget;
+    } else if (expression is IndexExpression) {
+      return expression.realTarget;
+    }
+    throw UnimplementedError('Unhandled ${expression.runtimeType}'
+        '($expression)');
+  }
+
+  /// Recursively insert [cascadeTarget] (the LHS of the cascade) into the
+  /// LHS of the assignment expression that used to be the cascade's RHS.
+  static Expression _insertCascadeTargetIntoExpression(
+      Expression expression, Expression cascadeTarget) {
+    // Base case: We've recursed as deep as possible.
+    if (expression == cascadeTarget) return cascadeTarget;
+
+    // Otherwise, copy `expression` and recurse into its LHS.
+    var expressionTarget = _realTargetOf(expression);
+    if (expression is PropertyAccess) {
+      return astFactory.propertyAccess(
+          _insertCascadeTargetIntoExpression(expressionTarget, cascadeTarget),
+          // If we've reached the end, replace the `..` operator with `.`
+          expressionTarget == cascadeTarget
+              ? _period(expression.operator)
+              : expression.operator,
+          expression.propertyName);
+    } else if (expression is MethodInvocation) {
+      return astFactory.methodInvocation(
+          _insertCascadeTargetIntoExpression(expressionTarget, cascadeTarget),
+          // If we've reached the end, replace the `..` operator with `.`
+          expressionTarget == cascadeTarget
+              ? _period(expression.operator)
+              : expression.operator,
+          expression.methodName,
+          expression.typeArguments,
+          expression.argumentList);
+    } else if (expression is IndexExpression) {
+      return astFactory.indexExpressionForTarget(
+          _insertCascadeTargetIntoExpression(expressionTarget, cascadeTarget),
+          expression.leftBracket,
+          expression.index,
+          expression.rightBracket);
+    }
+    throw UnimplementedError('Unhandled ${expression.runtimeType}'
+        '($expression)');
+  }
+
+  /// Parenthesize the target of the given statement's expression (assumed to
+  /// be a CascadeExpression) before removing the cascade.
+  void _fixCascadeByParenthesizingTarget(ExpressionStatement statement) {
+    CascadeExpression cascade = statement.expression;
+    assert(cascade.cascadeSections.length == 1);
+
+    // Write any leading comments and whitespace immediately, as they should
+    // precede the new opening parenthesis, but then prevent them from being
+    // written again after the parenthesis.
+    writePrecedingCommentsAndNewlines(cascade.target.beginToken);
+    _suppressPrecedingCommentsAndNewLines.add(cascade.target.beginToken);
```

Comment:
Instead of suppressing comments on the old token, is it feasible to detach them and reattach them to the new `(` token?